pkg/operator/e2e/main_test.go

```diff
@@ -764,7 +768,96 @@ func validateCollectorUpMetrics(ctx context.Context, t *testContext, job string)
 				metric.type = "prometheus.googleapis.com/up/gauge" AND
 				metric.labels.external_key = "external_val"
 				`,
-						projectID, location, cluster, t.namespace, job, pod.Name, port,
+						projectID, location, cluster, t.namespace, job, pod.Spec.NodeName, port,
+					),
+					Interval: &gcmpb.TimeInterval{
+						EndTime:   timestamppb.New(now),
+						StartTime: timestamppb.New(now.Add(-10 * time.Second)),
+					},
+				})
+				series, err := iter.Next()
+				if err == iterator.Done {
+					t.Logf("No data, retrying...")
+					return false, nil
+				} else if err != nil {
+					return false, errors.Wrap(err, "querying metrics failed")
+				}
+				if v := series.Points[len(series.Points)-1].Value.GetDoubleValue(); v != 1 {
+					t.Logf("Up still %v, retrying...", v)
+					return false, nil
+				}
+				// We expect exactly one result.
+				series, err = iter.Next()
+				if err != iterator.Done {
+					return false, errors.Errorf("expected iterator to be done but got error %q and series %v", err, series)
+				}
+				return true, nil
+			}, ctx.Done())
+			if err != nil {
+				t.Fatalf("Waiting for collector metrics to appear in Cloud Monitoring failed: %s", err)
+			}
+		}
+	}
+}
+
+// testCollectorSelfPodMonitoring sets up pod monitoring of the collector itself
+// and waits for samples to become available in Cloud Monitoring.
+func testCollectorScrapeKubelet(ctx context.Context, t *testContext) {
+	if skipGCM {
+		t.Log("Not validating scraping of kubelets when --skip-gcm is set")
+		return
+	}
+	if projectID == "" {
+		t.Fatalf("no project specified (--project-id flag)")
+	}
+	if location == "" {
+		t.Fatalf("no location specified (--location flag)")
+	}
+	if cluster == "" {
+		t.Fatalf("no cluster name specified (--cluster flag)")
+	}
+
+	// Wait for metric data to show up in Cloud Monitoring.
+	metricClient, err := gcm.NewMetricClient(ctx)
+	if err != nil {
+		t.Fatalf("Create GCM metric client: %s", err)
+	}
+	defer metricClient.Close()
+
+	pods, err := t.kubeClient.CoreV1().Pods(t.namespace).List(ctx, metav1.ListOptions{
```

Comment:
Here we're not explicitly interested in collector pods, but rather all nodes in the cluster. I understand the end result in the test logic below is the same, but to be explicit, can we change this to fetching via the kube node list api instead to make that clear?